src/source/src/manager.rs

```diff
@@ -147,7 +148,12 @@ impl SourceManager for MemSourceManager {
             "expected row_id_index >= 0, got {}",
             info.row_id_index
         );
-        let row_id_index = info.row_id_index as usize;
+        let row_id_index = if info.pk_column_ids != vec![0] {
+            None
+        } else {
+            Some(0)
+        };
```

Comment:
So we're going to ignore the `info.row_id_index`? I think this should be decided in the frontend.

Reply:
Frontend will do it later. @xiangjinwu says there may be some compatible issues.

Reply:
We can keep this implementation and make it work first, and then update `row_id_index` in proto as optional so that it is decided on frontend rather than handled here.